presto-spi/src/main/java/io/prestosql/spi/Page.java

```diff
@@ -318,6 +318,28 @@ public Page getPositions(int[] retainedPositions, int offset, int length)
         return wrapBlocksWithoutCopy(length, blocks);
     }
 
+    public Page copyPositions(int[] retainedPositions, int offset, int length)
+    {
+        requireNonNull(retainedPositions, "retainedPositions is null");
+
+        Block[] blocks = new Block[this.blocks.length];
+        for (int i = 0; i < blocks.length; i++) {
+            blocks[i] = this.blocks[i].copyPositions(retainedPositions, offset, length);
+        }
+        return wrapBlocksWithoutCopy(length, blocks);
+    }
+
+    public Page extractChannels(int[] channels)
```

Comment:
Let's name this `public Page getColumns(int... columns)`, which matches the other methods in this class better.

Reply:
Renamed method and accepted a varargs argument.